aspnetcore/fundamentals/static-files.md

```diff
@@ -1,17 +1,1150 @@
+
+* Complete the following steps in IIS Manager to remove the IIS static file handler at the server or website level:
+
+  1. Navigate to the **Modules** feature.
+  1. Select **StaticFileModule** in the list.
+  1. Click **Remove** in the **Actions** sidebar.
+
+> [!WARNING]
+> If the IIS static file handler is enabled **and** the ASP.NET Core Module is configured incorrectly, static files are served. This happens, for example, if the `web.config` file isn't deployed.
+
+* Place code files, including `.cs` and `.cshtml`, outside of the app project's [web root](xref:fundamentals/index#web-root). A logical separation is therefore created between the app's client-side content and server-based code. This prevents server-side code from being leaked.
+
+## MSBuild properties
+
+The following tables show the static files MSBuild properties and metadata descriptions.
+
+Property | Description
+--- | ---
+`EnableDefaultCompressedItems` | Enables default compression include/exclude patterns.
+`CompressionIncludePatterns` | Semicolon-separated list of file patterns to include for compression.
+`CompressionExcludePatterns` | Semicolon-separated list of file patterns to exclude from compression.
+`EnableDefaultCompressionFormats` | Enables default compression formats (Gzip and Brotli).
+`BuildCompressionFormats` | Compression formats to use during build.
+`PublishCompressionFormats` | Compression formats to use during publish.
+`DisableBuildCompression` | Disables compression during build.
+`CompressDiscoveredAssetsDuringBuild` | Compresses discovered assets during build.
+`BrotliCompressionLevel` | Compression level for the Brotli algorithm.
+`LinkAlternativeRepresentationsToOriginalResource` | How to link compressed assets to original resources.
+`StaticWebAssetBuildCompressAllAssets` | Compresses all assets during build, not just assets discovered or computed during a build.
+`StaticWebAssetPublishCompressAllAssets` | Compresses all assets during publish, not just assets discovered or computed during a build.
+
+Property | Description
+--- | ---
+`StaticWebAssetBasePath` | Base URL path for all the assets in a library.
+`StaticWebAssetsFingerprintContent` | Enables content fingerprinting for cache busting.
+`StaticWebAssetFingerprintingEnabled` | Enables fingerprinting feature for static web assets.
+`StaticWebAssetsCacheDefineStaticWebAssetsEnabled` | Enables caching for static web asset definitions.
+`StaticWebAssetBuildManifestPath` | Path to the build manifest JSON file.
+`StaticWebAssetsBuildManifestCacheFilePath` | Path to the build manifest cache file.
+`StaticWebAssetEndpointExclusionPattern` | Pattern for excluding endpoints.
+
+Item group | Description | Metadata
+--- | ---
+`StaticWebAssetContentTypeMapping` | Maps file patterns to content types and cache headers for endpoints. | Pattern, Cache
```

Comment:
This should be covered under a section about "adding new content type mappings at build time"

Reply:
Ok ... I have one more pass to get the sample app out of here. It's in the samples repo now. I didn't do it yet because I didn't want to clutter the diff. I'll take care of it by EOD.